dashboard/modules/job/job_head.py

```diff
@@ -177,70 +230,153 @@ async def stop_job(self, req: Request) -> Response:
             text=json.dumps(dataclasses.asdict(resp)), content_type="application/json"
         )
 
-    @routes.get("/api/jobs/{job_id}")
+    @routes.get("/api/jobs/{job_or_submission_id}")
     @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=False)
     async def get_job_info(self, req: Request) -> Response:
-        job_id = req.match_info["job_id"]
-        if not self.job_exists(job_id):
+        job_or_submission_id = req.match_info["job_or_submission_id"]
+        job = await self.find_job_by_ids(job_or_submission_id)
+        if not job:
             return Response(
-                text=f"Job {job_id} does not exist",
+                text=f"Job {job_or_submission_id} does not exist",
                 status=aiohttp.web.HTTPNotFound.status_code,
             )
 
-        data: JobInfo = self._job_manager.get_job_info(job_id)
         return Response(
-            text=json.dumps(dataclasses.asdict(data)), content_type="application/json"
+            text=json.dumps(job.dict()),
+            content_type="application/json",
         )
 
     @routes.get("/api/jobs/")
     @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=False)
     async def list_jobs(self, req: Request) -> Response:
-        data: dict[str, JobInfo] = self._job_manager.list_jobs()
+        driver_jobs, submission_job_drivers = await self._get_driver_jobs()
+
+        # TODO(aguo): convert _job_manager.list_jobs to an async function.
+        submission_jobs = self._job_manager.list_jobs()
+        submission_jobs = [
+            JobDetails(
+                **dataclasses.asdict(job),
+                submission_id=submission_id,
+                job_id=submission_job_drivers.get(submission_id).id
+                if submission_id in submission_job_drivers
+                else None,
+                driver_info=submission_job_drivers.get(submission_id),
+                type=JobType.SUBMISSION,
+            )
+            for submission_id, job in submission_jobs.items()
+        ]
         return Response(
             text=json.dumps(
-                {
-                    job_id: dataclasses.asdict(job_info)
-                    for job_id, job_info in data.items()
-                }
+                [
+                    *[submission_job.dict() for submission_job in submission_jobs],
+                    *[job_info.dict() for job_info in driver_jobs.values()],
+                ]
             ),
             content_type="application/json",
         )
 
-    @routes.get("/api/jobs/{job_id}/logs")
+    async def _get_driver_jobs(self):
+        """Returns a tuple of dictionaries related to drivers.
+
+        The first dictionary contains all driver jobs and is keyed by the job's id.
```

Comment:
Why don't we use a named tuple as a return value?

Reply:
hmm actually why do we need to distinguish them at all? maybe we can return both types of jobs to the same data structure?

Reply:
one is keyed by job_id and the other is keyed by submission_id for easier lookup.